src/main/java/org/example/dao/ShortestPathDAOImpl.java

```diff
@@ -0,0 +1,73 @@
+package org.example.dao;
+
+import org.example.configurations.ConnectionPool;
+import org.example.model.Point;
+import org.example.model.ShortestPath;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+public class ShortestPathDAOImpl implements ShortestPathDAO {
+    private static final ConnectionPool CONNECTIONPOOL = ConnectionPool.getInstance();
+
+    public ShortestPathDAOImpl() {
+    }
+
+    @Override
+    public void insertShortestPath(ShortestPath shortestPath) {
+        try (Connection connection = CONNECTIONPOOL.getConnectionFromPool();
```

Comment:
I don't think the connection should be created in the try with resources. Use a finally block